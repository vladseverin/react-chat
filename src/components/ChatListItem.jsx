import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import { withStyles } from '@material-ui/core/styles';
import { Link } from 'react-router-dom';
import red from '@material-ui/core/colors/red';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Avatar from './Avatar';

const styles = () => ({
  activeItem: {
    backgroundColor: red[50],
  },
});

const ChatListItem = ({
  classes, disabled, title, chatId, active, createdAt,
}) => (
  <ListItem
    button
    component={Link}
    to={`/chat/${chatId}`}
    className={active ? classes.activeItem : ''}
    disabled={disabled}
  >
    <Avatar colorFrom={chatId}>{title}</Avatar>
    <ListItemText primary={title} secondary={moment(createdAt).fromNow()} />
  </ListItem>
);

ChatListItem.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  disabled: PropTypes.bool.isRequired,
  title: PropTypes.string,
  chatId: PropTypes.string.isRequired,
  active: PropTypes.bool,
  createdAt: PropTypes.string.isRequired,
};

ChatListItem.defaultProps = {
  active: null,
  title: '',
};

export default withStyles(styles)(ChatListItem);
