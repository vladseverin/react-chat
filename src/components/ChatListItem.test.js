import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import ChatListItem from './ChatListItem';

jest.mock('moment', () => () => ({ fromNow: () => 'a few seconds ago' }));

const mockProps = {
  disabled: false,
  active: false,
  chatId: '12345',
  title: 'myChat',
  createdAt: '2018-06-13T20:48:03.200Z',
};

describe('<ChatListItem />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <ChatListItem {...mockProps} />
      </MemoryRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter>
          <ChatListItem {...mockProps} />
              </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
