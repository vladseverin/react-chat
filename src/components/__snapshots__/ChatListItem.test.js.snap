// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatListItem /> renders correctly 1`] = `
<a
  className="MuiButtonBase-root-66 MuiListItem-root-56 MuiListItem-default-59 MuiListItem-gutters-63 MuiListItem-button-64"
  href="/chat/12345"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  <div
    className="MuiAvatar-root-69 MuiAvatar-colorDefault-70"
    style={
      Object {
        "backgroundColor": "#d32f2f",
      }
    }
  >
    M
  </div>
  <div
    className="MuiListItemText-root-72"
  >
    <h3
      className="MuiTypography-root-78 MuiTypography-subheading-85 MuiListItemText-primary-75"
    >
      myChat
    </h3>
    <p
      className="MuiTypography-root-78 MuiTypography-body1-87 MuiTypography-colorTextSecondary-100 MuiListItemText-secondary-76"
    >
      a few seconds ago
    </p>
  </div>
  <span
    className="MuiTouchRipple-root-102"
  />
</a>
`;
